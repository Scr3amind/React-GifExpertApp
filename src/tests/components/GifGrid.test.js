import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid";
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

    test('debe de mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={'Goku'}/>);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://demo.url',
            title: 'titulo',

        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow(<GifGrid category={'Goku'}/>);
        
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
    
    
})
