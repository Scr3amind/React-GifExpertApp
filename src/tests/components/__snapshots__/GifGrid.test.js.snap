// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <p>
     Loading... 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
